flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/source/splitter/MongoShardedSplitterTest.java

```diff
@@ -0,0 +1,145 @@
+public class MongoShardedSplitterTest {
+
+    @Mock private MongoClient mongoClient;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testShardedSplitter() {
+        MongoNamespace namespace = new MongoNamespace("test_db.test_coll");
+        BsonDocument mockCollectionMetadata = mockCollectionMetadata();
+        ArrayList<BsonDocument> mockChunksData = mockChunksData();
+
+        MongoSplitContext splitContext =
+                MongoSplitContext.of(
+                        MongoReadOptions.builder().build(),
+                        mongoClient,
+                        namespace,
+                        mockCollStats());
+
+        List<MongoScanSourceSplit> expected = new ArrayList<>();
+        for (int i = 0; i < mockChunksData.size(); i++) {
+            BsonDocument mockChunkData = mockChunksData.get(i);
+            expected.add(
+                    new MongoScanSourceSplit(
+                            String.format("%s_%d", namespace, i),
+                            namespace.getDatabaseName(),
+                            namespace.getCollectionName(),
+                            mockChunkData.getDocument(MIN_FIELD),
+                            mockChunkData.getDocument(MAX_FIELD),
+                            mockCollectionMetadata.getDocument(KEY_FIELD)));
+        }
+
+        try (MockedStatic<MongoUtils> util = mockStatic(MongoUtils.class)) {
+            util.when(() -> MongoUtils.readCollectionMetadata(any(), any()))
+                    .thenReturn(mockCollectionMetadata);
+
+            util.when(() -> MongoUtils.readChunks(any(), any())).thenReturn(mockChunksData);
+
+            util.when(() -> MongoUtils.isValidShardedCollection(any())).thenReturn(true);
+
+            Collection<MongoScanSourceSplit> actual =
+                    MongoShardedSplitter.INSTANCE.split(splitContext);
+            assertThat(actual, equalTo(expected));
+        }
+    }
+
+    private BsonDocument mockCollectionMetadata() {
+        return new BsonDocument()
+                .append(ID_FIELD, new BsonObjectId())
+                .append(UUID_FIELD, new BsonBinary(UUID.randomUUID()))
+                .append(DROPPED_FIELD, BsonBoolean.FALSE)
+                .append(KEY_FIELD, ID_HINT);
+    }
+
+    private ArrayList<BsonDocument> mockChunksData() {
+        ArrayList<BsonDocument> chunks = new ArrayList<>();
+        chunks.add(mockChunkData(1));
+        chunks.add(mockChunkData(2));
+        chunks.add(mockChunkData(3));
+        return chunks;
+    }
+
+    private BsonDocument mockChunkData(int index) {
+        return new BsonDocument()
+                .append(MIN_FIELD, new BsonDocument(ID_FIELD, new BsonInt32(index * 100)))
+                .append(MAX_FIELD, new BsonDocument(ID_FIELD, new BsonInt32((index + 1) * 100)))
+                .append(SHARD_FIELD, new BsonString("shard-" + index));
+    }
+
+    private BsonDocument mockCollStats() {
```

Comment:
```suggestion
    private static BsonDocument mockCollStats() {
```
add static modifier when possible